nslsii/temperature_controllers.py

```diff
@@ -0,0 +1,80 @@
+from ophyd import DeviceStatus, Device, Component as Cpt, EpicsSignal, Signal
+
+
+class Eurotherm(Device):
+    '''This class is used for integrating with Eurotherm controllers.
+
+    This is used for Eurotherm controllers and is designed to ensure that the
+    set returns 'done' status only after the temperature has reached
+    equilibrium at the required value not when it first reaches the required
+    value. This is done via the attributes `self.equilibrium_time` and
+    `self.tolerance`. It only returns `done` if `self.readback` remains within
+    `self.tolerance` of `self.setpoint` over `self.equilibrium_time`. A third
+    attribute, `self.timeout`, is used to determeine the maximum time to wait
+    for equilibrium. If it takes longer than this it raises a TimeoutError.
+
+    Parameters
+    ----------
+    pv_prefix : str.
+        The PV prefix that is common to the readback and setpoint PV's.
+    '''
+
+    def __init__(self, pv_prefix, **kwargs):
+        super().__init__(pv_prefix, **kwargs)
+        self._set_lock = False
```

Comment:
We should maybe we using real locks for this but this is a common enough pattern in ophyd so I think we can circle back for that later.

Reply:
I am happy to circle back later.

Reply:
@danielballan just so I can circle back to this later, when you specify 'real lock' do you mean a threading.Lock instacne as described [here](https://docs.python.org/3.6/library/threading.html#timer-objects)?